docs/reference/notifications.md

```diff
@@ -0,0 +1,294 @@
+# Notifications
+
+_Introduced in InvenioRDM v12_
+
+The notification system in InvenioRDM is provided via the [`invenio-notifications`]() module.
+This module provides customization possibilities for notification backends, notification builders and resolvers.
+A notification manager is created, which will rely on these configuration values and provide logic to send notifications. Its task is also to call respective methods to resolve the notification context, generate recipients, filter recipients and generate the backend ids for further processing. With all information created, it will then dispatch further tasks to relay the notification to the backend, which will take care of sending the actual notification.
+
+Notifications are registered at the service level in the unit of work and send off to a celery task - which takes care of further processing.
+
+## Base Entities
+
+Let's first have a look at the base entities provided.
+
+### Models
+
+#### Notification
+
+A notification is a simple dataclass, holding information about its type and the context.
+
+```py
+@dataclass
+class Notification:
+
+    type: str
+    context: dict
+```
+
+The context attribute of the notification holds information relevant for further processing and will be expanded throughout the workflow.
+When dispatching this notification in the service, the context shall be as minimal as possible, to reduce passing huge chunks of data.
```

Comment:
> to reduce passing huge chunks of data.

Maybe add what are the implications in case that it was done (i.e. performance, etc.)

Reply:
Added implications of passing a large context/notification